송원/서울지하철2호선/index.py

```diff
@@ -0,0 +1,71 @@
+import sys
+from collections import deque
+sys.stdin = open("input.txt", "r")
+
+n = int(input())
+graph = [[] for _ in range(n + 1)]
+
+for _ in range(n):
+  a, b = map(int, input().split())
+  graph[a].append(b)
+  graph[b].append(a)
+
+visited = [0] * (n + 1)
+result = []
+
+def DFS(node, prev):
+  visited[node] = 1
+  found = False
+  for i in graph[node]:
+    if visited[i] == 0:
+      DFS(i, node)
+      if visited[i] == 2:
+        visited_count = 0
+        for j in graph[i]:
+          if visited[j] == 2:
+            visited_count += 1
+        if visited_count < 2:
+          visited[node] = 2
+    elif visited[i] == 1 and i != prev and prev != 0:
+      visited[i] = 2
+      visited[node] = 2
+      found = True
+  if found:
+    return True
+  else:
+    return False
+  
+  return False
```

Comment:
여기는 한줄로 줄일 수 있을 것 같아요!
```suggestion
  return found
```